### PR TITLE
bugfix/14417-incorrect-min-max-behaviour

### LINKED ISSUE #14417: yAxis min property and startOnTick/endOnTick not working properly
#### Expected behaviour

When I use startOnTick and endOnTick as false, and min property in yAxis, if that min value is greater than all the values of the points then chart should be empty (it shouldn't draw any point)

#### Actual behaviour

When I use startOnTick and endOnTick as false and a min property in yAxis, if the min value is greater than the values of the points, it actually draws some points (that are lower than the min value). Also, it seems that the chart is inverted

#### Live demo with steps to reproduce
https://jsfiddle.net/ladet/0k45zcdb/2/

#### Product version
Highcharts 8.1.2 with Highcharts-react

**Note**: I need both startOnTick and endOnTick as false, and it shouldn't be drawing any points when the min is greater than the points. If I set startOnTick and endOnTick as true, this doesn't happen. If I set a max property it works correctly, but since both max and min are set dynamically by the user, it should be possible to only select a minimum limit and it should be working properly whether the values are lower than said minimum or higher.

image

This also happens with columns charts but is worse since it draws a full column when it shouldn't draw any column
image

### PR DESCRIPTION
**Pull request by DJTechnoo**
Fixed #14417, charts displayed parts of the data when min or max exceeded extreme values.

~~Fixed #14417 charts displaying parts when min or max exceed extreme values.~~

### CODE REVIEW

File: ts/Core/Axis/Axis.ts [hunk 1]
```diff
@@ -5794,6 +5794,19 @@ class Axis implements AxisComposition, AxisLike {
                 }
             }
 
+            // If min is bigger than highest,
+            // or if max less than lowest value,
+            // the chart should not render points. (#14417)
+            if (this.min && this.max) {
+                if (this.min > this.max) {
+                    if (typeof this.options.min !== 'undefined' && this.options.min !== null) {
```
Comment (pawelfus) [inline ts/Core/Axis/Axis.ts:5802]: This looks now exactly as `Highcharts.defined()` implementation

File: ts/Core/Axis/Axis.ts [hunk 2]
```diff
@@ -5794,6 +5794,19 @@ class Axis implements AxisComposition, AxisLike {
                 }
             }
 
+            // If min is bigger than highest,
+            // or if max less than lowest value,
+            // the chart should not render points. (#14417)
+            if (this.min && this.max) {
```
Comment (pawelfus) [inline ts/Core/Axis/Axis.ts:5800]: This probably will fail when `this.min = 0` and `this.max = -1`?

**Review by pawelfus — COMMENTED** [review]
I know the solution here is based on my workaround, but I used `trimTicks` in the workaround because that was the easiest place to inject some code. Instead it might be better to fix issue where it happens (where `axis.min` and `axis.max` are being defined) - check `setTickInterval` method.

**Comment by DJTechnoo** [comment]
Thank you for the review Pawel! Yes you are right, I shall have one more go at it, thank you for advice and feedback!

File: ts/Core/Axis/Axis.ts [hunk 3]
```diff
@@ -5485,6 +5485,18 @@ class Axis implements AxisComposition, AxisLike {
             }
         }
 
+        // If min is bigger than highest,
+        // or if max less than lowest value,
+        // the chart should not render points. (#14417)
```
Comment (TorsteinHonsi) [inline ts/Core/Axis/Axis.ts:5490]: Let the comment break at line-width 80, so it looks less random.

File: ts/Core/Axis/Axis.ts [hunk 4]
```diff
@@ -5485,6 +5485,18 @@ class Axis implements AxisComposition, AxisLike {
...
+        if (axis.min !== null && axis.max !== null) {
+            if (axis.min > axis.max) {
```
Comment (TorsteinHonsi) [inline ts/Core/Axis/Axis.ts:5492]: These two conditions can be combined into one.

**Review by TorsteinHonsi — CHANGES_REQUESTED** [review]
Thanks! In addition to the two inline comments, please also [make the PR ready for the changelog](https://github.com/highcharts/highcharts/tree/master/changelog#writing-content-for-the-changelog).

File: ts/Core/Axis/Axis.ts (hunk 4 above)
Comment (DJTechnoo) [reply ts/Core/Axis/Axis.ts:5492]: Thank you Torstein, will do the changes and prepare the PR after resolving the failing tests that are occuring.

**Review by TorsteinHonsi — APPROVED** [review]
Thanks!

**Pull request merged**